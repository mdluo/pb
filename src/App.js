import React, { Component } from 'react';
import { Button } from 'antd';

import logo from './logo.svg';
import styles from './App.module.less';

class App extends Component {
  render() {
    return (
      <div className={styles.app}>
        <header className={styles.header}>
          <img src={logo} className={styles.logo} alt="logo" />
          <h1 className={styles.title}>Welcome to React</h1>
        </header>
        <p className={styles.intro}>
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
        <Button icon="star-o">Antd Button</Button>
      </div>
    );
  }
}

export default App;
