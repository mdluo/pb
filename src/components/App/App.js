import React, { Component } from 'react';
import { Select, Button, Row, Col } from 'antd';

import Page from 'components/Layout/Page';
import Bar from 'components/Bar';
import request from 'utils/request';

import styles from './App.module.less';

const { Option } = Select;

class App extends Component {
  state = {
    loading: true,
    active: null,
    bars: null,
    buttons: null,
    limit: null,
  }

  componentDidMount() {
    this.fetch();
  }

  fetch = async () => {
    const { data } = await request();
    if (data) {
      this.setState({
        active: 0,
        bars: data.bars.map((value, id) => ({ id, value })),
        buttons: data.buttons.map((value, id) => ({ id, value })),
        limit: data.limit,
        loading: false,
      });
    }
  }

  handleSelect = (active) => {
    this.setState({ active });
  }

  handleClick = (value) => {
    const { active, bars } = this.state;
    this.setState({
      bars: bars.map((bar) => {
        if (bar.id === active) {
          return {
            id: bar.id,
            value: Math.max(0, bar.value + value),
          };
        }
        return bar;
      }),
    });
  }

  render() {
    const { loading, active, bars, buttons, limit } = this.state;
    return (
      <Page loading={loading}>
        <p>Limit: <b>{limit}</b></p>
        <div className={styles.bars}>
          {bars && bars.map(({ id, value }) => (
            <Bar key={id} limit={limit} value={value} active={active === id} />
          ))}
        </div>
        <Row gutter={24}>
          <Col className={styles.select} xs={24} sm={8}>
            <Select style={{ width: '100%' }} value={active} onChange={this.handleSelect}>
              {bars && bars.map(({ id }) => (
                <Option key={id} value={id}>
                  # Progress {id + 1}
                </Option>
              ))}
            </Select>
          </Col>
          <Col className={styles.buttons} xs={24} sm={16}>
            {buttons && buttons.map(({ id, value }) => (
              <Button key={id} onClick={() => this.handleClick(value)}>
                {value >= 0 ? `+${value}` : value}
              </Button>
            ))}
          </Col>
        </Row>
      </Page>
    );
  }
}

export default App;
