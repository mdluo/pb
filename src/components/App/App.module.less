.bars {
  margin: 20px 0 25px;
}

.select {
  margin-bottom: 15px;
}

.buttons > button {
  margin: 0 10px 15px 0;
}

.option {
}
