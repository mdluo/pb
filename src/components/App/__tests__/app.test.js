import React from 'react';
import { render, mount } from 'enzyme';

import App from '..';

jest.mock('utils/request');

describe('App', () => {
  it('renders initial loading state', () => {
    const wrapper = render(<App />);
    expect(wrapper.find('.ant-card-loading-content')).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders loaded state', (done) => {
    const wrapper = mount(<App />);
    setTimeout(() => {
      wrapper.update();
      expect(wrapper).toMatchSnapshot();
      done();
    });
  });

  it('renders correct count of bars', (done) => {
    const wrapper = mount(<App />);
    expect(wrapper.find('Bar')).toHaveLength(0);
    setTimeout(() => {
      wrapper.update();
      expect(wrapper.find('Bar')).toHaveLength(3);
      done();
    });
  });

  it('renders correct count of options', (done) => {
    const wrapper = mount(<App />);
    setTimeout(() => {
      wrapper.update();
      wrapper.find('.ant-select').simulate('click');
      const dropdownWrapper = mount(wrapper.find('Trigger').instance().getComponent());
      expect(dropdownWrapper.find('MenuItem')).toHaveLength(3);
      done();
    });
  });

  it('renders correct count of buttons', (done) => {
    const wrapper = mount(<App />);
    expect(wrapper.find('Button')).toHaveLength(0);
    setTimeout(() => {
      wrapper.update();
      expect(wrapper.find('Button')).toHaveLength(4);
      done();
    });
  });

  it('active the first Bar automatically', (done) => {
    const wrapper = mount(<App />);
    setTimeout(() => {
      wrapper.update();
      expect(wrapper.find('Bar').first().props().active).toBe(true);
      done();
    });
  });

  it('active the correct Bar when selected', (done) => {
    const wrapper = mount(<App />);
    setTimeout(() => {
      wrapper.update();
      wrapper.find('.ant-select').simulate('click');
      const dropdownWrapper = mount(wrapper.find('Trigger').instance().getComponent());
      const menuItemWrapper = dropdownWrapper.find('MenuItem').at(1);
      menuItemWrapper.simulate('click');
      wrapper.update();
      expect(wrapper.find('Bar').at(1).props().active).toBe(true);
      done();
    });
  });

  it('update value to the active Bar when Button clicked', (done) => {
    const wrapper = mount(<App />);
    setTimeout(() => {
      wrapper.update();
      const buttonWrapper2 = wrapper.find('Button').at(2);
      const buttonWrapper3 = wrapper.find('Button').at(3);
      buttonWrapper3.simulate('click'); // -40
      wrapper.update();
      buttonWrapper2.simulate('click'); // +15
      wrapper.update();
      expect(wrapper.find('Bar').at(0).props().value).toBe(15);
      done();
    });
  });
});
