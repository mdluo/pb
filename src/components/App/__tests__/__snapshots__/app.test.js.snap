// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders initial loading state 1`] = `
<div
  class="layout ant-layout"
>
  <div
    class="ant-layout-content"
  >
    <div
      class="ant-card card ant-card-loading"
    >
      <div
        class="ant-card-head"
      >
        <div
          class="ant-card-head-wrapper"
        >
          <div
            class="ant-card-head-title"
          >
            Progress Bar Demo
          </div>
          <div
            class="ant-card-extra"
          >
            <a
              href="https://github.com/mdluo/pb"
              rel="noopener noreferrer"
              target="_blank"
            >
              <i
                class="anticon anticon-link"
              />
               GitHub
            </a>
          </div>
        </div>
      </div>
      <div
        class="ant-card-body"
      >
        <div
          class="ant-card-loading-content"
        >
          <div
            class="ant-row"
            style="margin-left:-4px;margin-right:-4px"
          >
            <div
              class="ant-col-22"
              style="padding-left:4px;padding-right:4px"
            >
              <div
                class="ant-card-loading-block"
              />
            </div>
          </div>
          <div
            class="ant-row"
            style="margin-left:-4px;margin-right:-4px"
          >
            <div
              class="ant-col-8"
              style="padding-left:4px;padding-right:4px"
            >
              <div
                class="ant-card-loading-block"
              />
            </div>
            <div
              class="ant-col-15"
              style="padding-left:4px;padding-right:4px"
            >
              <div
                class="ant-card-loading-block"
              />
            </div>
          </div>
          <div
            class="ant-row"
            style="margin-left:-4px;margin-right:-4px"
          >
            <div
              class="ant-col-6"
              style="padding-left:4px;padding-right:4px"
            >
              <div
                class="ant-card-loading-block"
              />
            </div>
            <div
              class="ant-col-18"
              style="padding-left:4px;padding-right:4px"
            >
              <div
                class="ant-card-loading-block"
              />
            </div>
          </div>
          <div
            class="ant-row"
            style="margin-left:-4px;margin-right:-4px"
          >
            <div
              class="ant-col-13"
              style="padding-left:4px;padding-right:4px"
            >
              <div
                class="ant-card-loading-block"
              />
            </div>
            <div
              class="ant-col-9"
              style="padding-left:4px;padding-right:4px"
            >
              <div
                class="ant-card-loading-block"
              />
            </div>
          </div>
          <div
            class="ant-row"
            style="margin-left:-4px;margin-right:-4px"
          >
            <div
              class="ant-col-4"
              style="padding-left:4px;padding-right:4px"
            >
              <div
                class="ant-card-loading-block"
              />
            </div>
            <div
              class="ant-col-3"
              style="padding-left:4px;padding-right:4px"
            >
              <div
                class="ant-card-loading-block"
              />
            </div>
            <div
              class="ant-col-16"
              style="padding-left:4px;padding-right:4px"
            >
              <div
                class="ant-card-loading-block"
              />
            </div>
          </div>
          <div
            class="ant-row"
            style="margin-left:-4px;margin-right:-4px"
          >
            <div
              class="ant-col-8"
              style="padding-left:4px;padding-right:4px"
            >
              <div
                class="ant-card-loading-block"
              />
            </div>
            <div
              class="ant-col-6"
              style="padding-left:4px;padding-right:4px"
            >
              <div
                class="ant-card-loading-block"
              />
            </div>
            <div
              class="ant-col-8"
              style="padding-left:4px;padding-right:4px"
            >
              <div
                class="ant-card-loading-block"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-layout-footer"
  >
    <p>
      Powered by 
      <a
        href="https://github.com/facebook/create-react-app"
        rel="noopener noreferrer"
        target="_blank"
      >
        CRA
      </a>
      , 
      <a
        href="https://ant.design/"
        rel="noopener noreferrer"
        target="_blank"
      >
        Antd
      </a>
       and more.
    </p>
    <p>
      Made with 
      <i
        class="anticon anticon-heart heart"
      />
       by 
      <a
        href="https://github.com/mdluo"
        rel="noopener noreferrer"
        target="_blank"
      >
        mdluo
      </a>
      .
    </p>
  </div>
</div>
`;

exports[`App renders loaded state 1`] = `
<App>
  <Page
    loading={false}
  >
    <Adapter
      className="layout"
    >
      <BasicLayout
        className="layout"
        prefixCls="ant-layout"
      >
        <div
          className="layout ant-layout"
        >
          <Adapter>
            <Basic
              prefixCls="ant-layout-content"
            >
              <div
                className="ant-layout-content"
              >
                <Card
                  bordered={false}
                  className="card"
                  extra={
                    <Link
                      href="https://github.com/mdluo/pb"
                    >
                      <Icon
                        type="link"
                      />
                       GitHub
                    </Link>
                  }
                  loading={false}
                  title="Progress Bar Demo"
                >
                  <div
                    className="ant-card card"
                  >
                    <div
                      className="ant-card-head"
                    >
                      <div
                        className="ant-card-head-wrapper"
                      >
                        <div
                          className="ant-card-head-title"
                        >
                          Progress Bar Demo
                        </div>
                        <div
                          className="ant-card-extra"
                        >
                          <Link
                            href="https://github.com/mdluo/pb"
                          >
                            <a
                              href="https://github.com/mdluo/pb"
                              rel="noopener noreferrer"
                              target="_blank"
                            >
                              <Icon
                                type="link"
                              >
                                <i
                                  className="anticon anticon-link"
                                />
                              </Icon>
                               GitHub
                            </a>
                          </Link>
                        </div>
                      </div>
                    </div>
                    <div
                      className="ant-card-body"
                      style={Object {}}
                    >
                      <p>
                        Limit: 
                        <b>
                          120
                        </b>
                      </p>
                      <div
                        className="bars"
                      >
                        <Bar
                          active={true}
                          height={32}
                          key="0"
                          limit={120}
                          value={30}
                        >
                          <div
                            className="bar active"
                          >
                            <Progress
                              percent={25}
                              prefixCls="ant-progress"
                              showInfo={false}
                              size="default"
                              status={null}
                              strokeWidth={32}
                              trailColor="#f3f3f3"
                              type="line"
                            >
                              <div
                                className="ant-progress ant-progress-line ant-progress-status-normal ant-progress-default"
                              >
                                <div>
                                  <div
                                    className="ant-progress-outer"
                                  >
                                    <div
                                      className="ant-progress-inner"
                                    >
                                      <div
                                        className="ant-progress-bg"
                                        style={
                                          Object {
                                            "height": 32,
                                            "width": "25%",
                                          }
                                        }
                                      />
                                    </div>
                                  </div>
                                </div>
                              </div>
                            </Progress>
                            <div
                              className="reading"
                              style={
                                Object {
                                  "lineHeight": "32px",
                                }
                              }
                            >
                              <span
                                className="percentage"
                              >
                                25
                                %
                              </span>
                              <span
                                className="value"
                              >
                                (
                                30
                                )
                              </span>
                            </div>
                          </div>
                        </Bar>
                        <Bar
                          active={false}
                          height={32}
                          key="1"
                          limit={120}
                          value={60}
                        >
                          <div
                            className="bar"
                          >
                            <Progress
                              percent={50}
                              prefixCls="ant-progress"
                              showInfo={false}
                              size="default"
                              status={null}
                              strokeWidth={32}
                              trailColor="#f3f3f3"
                              type="line"
                            >
                              <div
                                className="ant-progress ant-progress-line ant-progress-status-normal ant-progress-default"
                              >
                                <div>
                                  <div
                                    className="ant-progress-outer"
                                  >
                                    <div
                                      className="ant-progress-inner"
                                    >
                                      <div
                                        className="ant-progress-bg"
                                        style={
                                          Object {
                                            "height": 32,
                                            "width": "50%",
                                          }
                                        }
                                      />
                                    </div>
                                  </div>
                                </div>
                              </div>
                            </Progress>
                            <div
                              className="reading"
                              style={
                                Object {
                                  "lineHeight": "32px",
                                }
                              }
                            >
                              <span
                                className="percentage"
                              >
                                50
                                %
                              </span>
                              <span
                                className="value"
                              >
                                (
                                60
                                )
                              </span>
                            </div>
                          </div>
                        </Bar>
                        <Bar
                          active={false}
                          height={32}
                          key="2"
                          limit={120}
                          value={80}
                        >
                          <div
                            className="bar"
                          >
                            <Progress
                              percent={66}
                              prefixCls="ant-progress"
                              showInfo={false}
                              size="default"
                              status={null}
                              strokeWidth={32}
                              trailColor="#f3f3f3"
                              type="line"
                            >
                              <div
                                className="ant-progress ant-progress-line ant-progress-status-normal ant-progress-default"
                              >
                                <div>
                                  <div
                                    className="ant-progress-outer"
                                  >
                                    <div
                                      className="ant-progress-inner"
                                    >
                                      <div
                                        className="ant-progress-bg"
                                        style={
                                          Object {
                                            "height": 32,
                                            "width": "66%",
                                          }
                                        }
                                      />
                                    </div>
                                  </div>
                                </div>
                              </div>
                            </Progress>
                            <div
                              className="reading"
                              style={
                                Object {
                                  "lineHeight": "32px",
                                }
                              }
                            >
                              <span
                                className="percentage"
                              >
                                66
                                %
                              </span>
                              <span
                                className="value"
                              >
                                (
                                80
                                )
                              </span>
                            </div>
                          </div>
                        </Bar>
                      </div>
                      <Row
                        gutter={24}
                      >
                        <div
                          className="ant-row"
                          style={
                            Object {
                              "marginLeft": -12,
                              "marginRight": -12,
                            }
                          }
                        >
                          <Col
                            className="select"
                            key=".0"
                            sm={8}
                            style={
                              Object {
                                "paddingLeft": 12,
                                "paddingRight": 12,
                              }
                            }
                            xs={24}
                          >
                            <div
                              className="select ant-col-xs-24 ant-col-sm-8"
                              style={
                                Object {
                                  "paddingLeft": 12,
                                  "paddingRight": 12,
                                }
                              }
                            >
                              <Select
                                choiceTransitionName="zoom"
                                onChange={[Function]}
                                prefixCls="ant-select"
                                showSearch={false}
                                style={
                                  Object {
                                    "width": "100%",
                                  }
                                }
                                transitionName="slide-up"
                                value={0}
                              >
                                <LocaleReceiver
                                  componentName="Select"
                                  defaultLocale={
                                    Object {
                                      "notFoundContent": "Not Found",
                                    }
                                  }
                                >
                                  <Select
                                    allowClear={false}
                                    autoClearSearchValue={true}
                                    backfill={false}
                                    choiceTransitionName="zoom"
                                    className=""
                                    combobox={false}
                                    defaultActiveFirstOption={true}
                                    defaultOpen={false}
                                    dropdownMatchSelectWidth={true}
                                    dropdownMenuStyle={Object {}}
                                    dropdownStyle={Object {}}
                                    labelInValue={false}
                                    multiple={false}
                                    notFoundContent="Not Found"
                                    onBlur={[Function]}
                                    onChange={[Function]}
                                    onDeselect={[Function]}
                                    onFocus={[Function]}
                                    onInputKeyDown={[Function]}
                                    onSearch={[Function]}
                                    onSelect={[Function]}
                                    optionFilterProp="value"
                                    optionLabelProp="children"
                                    placeholder=""
                                    prefixCls="ant-select"
                                    showAction={
                                      Array [
                                        "click",
                                      ]
                                    }
                                    showArrow={true}
                                    showSearch={false}
                                    style={
                                      Object {
                                        "width": "100%",
                                      }
                                    }
                                    tags={false}
                                    tokenSeparators={Array []}
                                    transitionName="slide-up"
                                    value={0}
                                  >
                                    <SelectTrigger
                                      combobox={false}
                                      defaultActiveFirstOption={true}
                                      dropdownMatchSelectWidth={true}
                                      dropdownMenuStyle={Object {}}
                                      dropdownStyle={Object {}}
                                      inputValue=""
                                      multiple={false}
                                      onDropdownVisibleChange={[Function]}
                                      onMenuDeselect={[Function]}
                                      onMenuSelect={[Function]}
                                      onPopupFocus={[Function]}
                                      options={Array []}
                                      prefixCls="ant-select"
                                      showAction={
                                        Array [
                                          "click",
                                        ]
                                      }
                                      showSearch={false}
                                      transitionName="slide-up"
                                      value={
                                        Array [
                                          0,
                                        ]
                                      }
                                      visible={false}
                                    >
                                      <Trigger
                                        action={Array []}
                                        afterPopupVisibleChange={[Function]}
                                        blurDelay={0.15}
                                        builtinPlacements={
                                          Object {
                                            "bottomLeft": Object {
                                              "offset": Array [
                                                0,
                                                4,
                                              ],
                                              "overflow": Object {
                                                "adjustX": 0,
                                                "adjustY": 1,
                                              },
                                              "points": Array [
                                                "tl",
                                                "bl",
                                              ],
                                            },
                                            "topLeft": Object {
                                              "offset": Array [
                                                0,
                                                -4,
                                              ],
                                              "overflow": Object {
                                                "adjustX": 0,
                                                "adjustY": 1,
                                              },
                                              "points": Array [
                                                "bl",
                                                "tl",
                                              ],
                                            },
                                          }
                                        }
                                        combobox={false}
                                        defaultActiveFirstOption={true}
                                        defaultPopupVisible={false}
                                        destroyPopupOnHide={false}
                                        dropdownMatchSelectWidth={true}
                                        dropdownMenuStyle={Object {}}
                                        dropdownStyle={Object {}}
                                        focusDelay={0}
                                        getDocument={[Function]}
                                        getPopupClassNameFromAlign={[Function]}
                                        hideAction={
                                          Array [
                                            "click",
                                          ]
                                        }
                                        inputValue=""
                                        mask={false}
                                        maskClosable={true}
                                        mouseEnterDelay={0}
                                        mouseLeaveDelay={0.1}
                                        multiple={false}
                                        onDropdownVisibleChange={[Function]}
                                        onMenuDeselect={[Function]}
                                        onMenuSelect={[Function]}
                                        onPopupAlign={[Function]}
                                        onPopupVisibleChange={[Function]}
                                        options={Array []}
                                        popup={
                                          <DropdownMenu
                                            backfillValue={undefined}
                                            defaultActiveFirstOption={true}
                                            dropdownMenuStyle={Object {}}
                                            firstActiveValue={undefined}
                                            inputValue=""
                                            menuItems={Array []}
                                            multiple={false}
                                            onMenuDeselect={[Function]}
                                            onMenuSelect={[Function]}
                                            onPopupFocus={[Function]}
                                            onPopupScroll={undefined}
                                            prefixCls="ant-select-dropdown"
                                            value={
                                              Array [
                                                0,
                                              ]
                                            }
                                            visible={false}
                                          />
                                        }
                                        popupAlign={Object {}}
                                        popupClassName="ant-select-dropdown--single"
                                        popupPlacement="bottomLeft"
                                        popupStyle={Object {}}
                                        popupTransitionName="slide-up"
                                        popupVisible={false}
                                        prefixCls="ant-select-dropdown"
                                        showAction={
                                          Array [
                                            "click",
                                          ]
                                        }
                                        showSearch={false}
                                        transitionName="slide-up"
                                        value={
                                          Array [
                                            0,
                                          ]
                                        }
                                        visible={false}
                                      >
                                        <div
                                          className="ant-select ant-select-enabled"
                                          key="trigger"
                                          onBlur={[Function]}
                                          onClick={[Function]}
                                          onFocus={[Function]}
                                          onMouseDown={[Function]}
                                          onTouchStart={[Function]}
                                          style={
                                            Object {
                                              "width": "100%",
                                            }
                                          }
                                        >
                                          <div
                                            aria-autocomplete="list"
                                            aria-expanded={false}
                                            aria-haspopup="true"
                                            className="ant-select-selection
            ant-select-selection--single"
                                            key="selection"
                                            onKeyDown={[Function]}
                                            role="combobox"
                                            tabIndex={0}
                                          >
                                            <div
                                              className="ant-select-selection__rendered"
                                            >
                                              <div
                                                className="ant-select-selection-selected-value"
                                                key="value"
                                                style={
                                                  Object {
                                                    "display": "block",
                                                    "opacity": 1,
                                                  }
                                                }
                                                title={null}
                                              >
                                                # Progress 
                                                1
                                              </div>
                                            </div>
                                            <span
                                              className="ant-select-arrow"
                                              key="arrow"
                                              onClick={[Function]}
                                              style={
                                                Object {
                                                  "WebkitUserSelect": "none",
                                                  "userSelect": "none",
                                                }
                                              }
                                              unselectable="on"
                                            >
                                              <b />
                                            </span>
                                          </div>
                                        </div>
                                      </Trigger>
                                    </SelectTrigger>
                                  </Select>
                                </LocaleReceiver>
                              </Select>
                            </div>
                          </Col>
                          <Col
                            className="buttons"
                            key=".1"
                            sm={16}
                            style={
                              Object {
                                "paddingLeft": 12,
                                "paddingRight": 12,
                              }
                            }
                            xs={24}
                          >
                            <div
                              className="buttons ant-col-xs-24 ant-col-sm-16"
                              style={
                                Object {
                                  "paddingLeft": 12,
                                  "paddingRight": 12,
                                }
                              }
                            >
                              <Button
                                ghost={false}
                                key="0"
                                loading={false}
                                onClick={[Function]}
                                prefixCls="ant-btn"
                              >
                                <button
                                  className="ant-btn"
                                  onClick={[Function]}
                                  type="button"
                                >
                                  <span
                                    key=".0"
                                  >
                                    +10
                                  </span>
                                </button>
                              </Button>
                              <Button
                                ghost={false}
                                key="1"
                                loading={false}
                                onClick={[Function]}
                                prefixCls="ant-btn"
                              >
                                <button
                                  className="ant-btn"
                                  onClick={[Function]}
                                  type="button"
                                >
                                  -20
                                </button>
                              </Button>
                              <Button
                                ghost={false}
                                key="2"
                                loading={false}
                                onClick={[Function]}
                                prefixCls="ant-btn"
                              >
                                <button
                                  className="ant-btn"
                                  onClick={[Function]}
                                  type="button"
                                >
                                  <span
                                    key=".0"
                                  >
                                    +15
                                  </span>
                                </button>
                              </Button>
                              <Button
                                ghost={false}
                                key="3"
                                loading={false}
                                onClick={[Function]}
                                prefixCls="ant-btn"
                              >
                                <button
                                  className="ant-btn"
                                  onClick={[Function]}
                                  type="button"
                                >
                                  -40
                                </button>
                              </Button>
                            </div>
                          </Col>
                        </div>
                      </Row>
                    </div>
                  </div>
                </Card>
              </div>
            </Basic>
          </Adapter>
          <Adapter>
            <Basic
              prefixCls="ant-layout-footer"
            >
              <div
                className="ant-layout-footer"
              >
                <p>
                  Powered by 
                  <Link
                    href="https://github.com/facebook/create-react-app"
                  >
                    <a
                      href="https://github.com/facebook/create-react-app"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      CRA
                    </a>
                  </Link>
                  , 
                  <Link
                    href="https://ant.design/"
                  >
                    <a
                      href="https://ant.design/"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      Antd
                    </a>
                  </Link>
                   and more.
                </p>
                <p>
                  Made with 
                  <Icon
                    className="heart"
                    type="heart"
                  >
                    <i
                      className="anticon anticon-heart heart"
                    />
                  </Icon>
                   by 
                  <Link
                    href="https://github.com/mdluo"
                  >
                    <a
                      href="https://github.com/mdluo"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      mdluo
                    </a>
                  </Link>
                  .
                </p>
              </div>
            </Basic>
          </Adapter>
        </div>
      </BasicLayout>
    </Adapter>
  </Page>
</App>
`;
